specutils/analysis/moment.py

```diff
@@ -64,9 +76,12 @@ def _compute_moment(spectrum, regions=None, order=0, axis=-1):
         return np.sum(flux, axis=axis)
 
     dispersion = spectral_axis
+    # We now have to account for the spectral axis being anywhere, not always last
     if len(flux.shape) > len(spectral_axis.shape):
-        _shape = flux.shape[:-1] + (1,)
-        dispersion = np.tile(spectral_axis, _shape)
+        for i in range(len(flux.shape)):
```

Comment:
I think this is the more idiomatic approach for numpy arrays
```suggestion
        for i in range(flux.ndim):
```